### PR TITLE
🏃 Update support for CAPI v0.3.5

### PR DESCRIPTION
**Pull request by prankul88**
**What this PR does / why we need it**:
Updated CAPI to [v0.3.5](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v0.3.5).

Also updated controller-gen tools to v0.2.9

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Update Cluster API to v0.3.5
```

### CODE REVIEW

**Comment by jichenjc** [comment]
looks good ~, let's see whether CI can pass

**Comment by prankul88** [comment]
Will check again

**Comment by prankul88** [comment]
Forgot to run `make modules`. Let's see if it works now.

File: Makefile [hunk 1]
```diff
@@ -335,7 +335,7 @@ create-cluster: $(CLUSTERCTL) $(KUSTOMIZE) $(ENVSUBST) ## Create a development K
 
 	# (Re-)install Core providers
 	$(CLUSTERCTL) delete --all
-	$(CLUSTERCTL) init --core cluster-api:v0.3.3 --bootstrap kubeadm:v0.3.3 --control-plane kubeadm:v0.3.3
+	$(CLUSTERCTL) init --core cluster-api:v0.3.5 --bootstrap kubeadm:v0.3.5 --control-plane kubeadm:v0.3.5
```
Comment (sbueringer) [inline Makefile:338]: we also have to update go.mod to 0.3.5

Comment (sbueringer) [reply Makefile:338]: We should also check if hack/tools/go uses the correct version of controller-tools and cluster-api/hack/tools (compare with CAPA when it already uses 0.3.5)

**Comment by sbueringer** [comment]
@prankul88 sorry for the flaky e2e test. Trying to get it fixed here: https://github.com/theopenlab/openlab/issues/504

File: Makefile (hunk 1 above)
Comment (prankul88) [reply Makefile:338]: @sbueringer I feel I have already updated the go.mod to 0.3.5 and also the corresponding controller tools in hack/tools. Would you mind elaborating a bit more?

Thanks :)

**Comment by prankul88** [comment]
> @prankul88 sorry for the flaky e2e test. Trying to get it fixed here: [theopenlab/openlab#504](https://github.com/theopenlab/openlab/issues/504)

No problem. Thanks for pointing it out, I was unaware of the ongoing issue.

File: Makefile (hunk 1 above)
Comment (sbueringer) [reply Makefile:338]: Sorry somehow missed :). The only thing I'm not sure about is cluster-api/hack/tools. Is that one also already the correct one?

Comment (prankul88) [reply Makefile:338]: @sbueringer I have updated the controller tools version to v0.2.9 in hack/tools. I think that is the correct version. I checked with CAPA repo which too uses that version.

Comment (sbueringer) [reply Makefile:338]: I was referring to the dependency with that name :)
See changes in https://github.com/kubernetes-sigs/cluster-api-provider-aws/commit/0580fa23476bbd797ecb5b3d1be218077f801662 (to hack/tools/go.mod)

They also upgraded to:
* k8s.io/code-generator v0.18.0
* sigs.k8s.io/cluster-api/hack/tools v0.0.0-20200430151019-59a696e07b5e

Comment (prankul88) [reply Makefile:338]: @sbueringer Ohh okay. Ya, I feel that needs to be updated too.

**Comment by prankul88** [comment]
@sbueringer PTAL

**Comment by jichenjc** [comment]
code looks good, seems the CI is not related..

**Comment by sbueringer** [comment]
recheck

**Pull request merged**